library/pathname/empty_spec.rb

```diff
@@ -0,0 +1,28 @@
+require File.expand_path('../../spec_helper', __dir__)
+require 'pathname'
+
+describe "Pathname#empty?" do
+  ruby_version_is "2.4" do
+    it 'returns true when file is not empty' do
+      (Pathname.new(__FILE__).empty?).should be_false
+    end
+
+    it 'returns false when the directory is not empty' do
+      (Pathname.new(__dir__).empty?).should be_false
+    end
+
+    it 'return true when file is empty' do
+      file = tmp('new_file_path_name.txt')
+      touch(file)
+      (Pathname.new(file).empty?).should be_true
+      File.delete(file)
```

Comment:
The `rm_r` helper is preferred to manually deleting a file/directory (notably because it does not allow deleting outside the `rubyspec_temp` directory).
Also, it is better to run the setup/teardown in `before :each`/`after :each` blocks, so that in the case Pathname#empty? fails, we still remove the file before going on to next specs.